### PR TITLE
Added function to split int array to array of ranges.

### PR DESCRIPTION
**Pull request by sergeymaksimov**
https://trello.com/c/NZO5SHSD
https://github.com/area9innovation/innovation/pull/2662
https://git.area9lyceum.com/Lyceum/lyceum/pulls/4164

### CODE REVIEW

File: lib/flowutils.flow [hunk 1]
```diff
@@ -37,11 +44,45 @@ isTest() -> bool {
...
+		ranges = mapiConcat(sorted, \i, id -> {
+			if (i > 0 && i < l && id - 1 == sorted[i - 1] && id + 1 == sorted[i + 1]) {
+				[]
+			} else if ((i == 0 || id - 1 != sorted[i - 1]) && i < l && id + 1 != sorted[i + 1]) {
+				[id, id]
+			} else {
+				[id]
+			}
+		});
+		if (length(ranges) < length(ids)) {
```
Comment (Eugeny48) [inline lib/flowutils.flow:73]: There is more cheap way to estimate: `if (foldi(ids, false, \i, acc, id -> acc || (i > 0 && ids[i - 1] + 1 == id)))`

File: lib/flowutils.flow [hunk 2]
```diff
@@ -37,11 +44,45 @@ isTest() -> bool {
...
-}
\ No newline at end of file
+}
+
+idsArrayToRanges(ids : [int]) -> RequestIds {
+	if (ids != [] && isUrlParameterTrue("load_by_range")) {
+		sorted = sort(ids);
+		l = length(sorted);
+		ranges = mapiConcat(sorted, \i, id -> {
+			if (i > 0 && i < l && id - 1 == sorted[i - 1] && id + 1 == sorted[i + 1]) {
```
Comment (Eugeny48) [inline lib/flowutils.flow:65]: `i < l` is `true` always

**Review by Eugeny48 — CHANGES_REQUESTED** [review]
Try to use fold + List + list2array. The current implementation is very expensive

File: lib/flowutils.flow [hunk 3]
```diff
@@ -20,9 +21,15 @@ export {
 	// For value == None it returns []
 	// For other values it returns [KeyValue(toStringNoExtraQuotes(value))]
 	buildKeyValueForPhp(key : string, value : ?) -> [KeyValue];
+	idsArrayToRanges(ids : [int]) -> RequestIds;
+	RequestIds ::= RequestIdsArray, RequestIdsRanges;
+		RequestIdsArray(ids : [int]);
+		RequestIdsRanges(ranges : [[int]]);
```
Comment (Eugeny48) [inline lib/flowutils.flow:27]: Please, add a comment explaining what `RequestIdsRanges` contains.

**Comment by sergeymaksimov** [comment]
> This way original string is even shorter. This rage should be "[[1, 2], 4, 6, 8]". No need for brackets around a single number, right?

Right.

**Review by Eugeny48 — APPROVED** [review]
Looks good.

**Pull request merged**